components/eam/src/physics/crm/samxx/crm_variance_transport.cpp

```diff
@@ -0,0 +1,286 @@
+      int ii = i + 2*(filter_wn_max+1) ;
+      int jj = j + 2*(filter_wn_max+1) ;
+      fft_out(k,jj,ii,icrm) = 0.0;
+    });
+  } else {
+    // for (int k=0; k<nzm; k++) {
+    //   for (int i=0; i<nwx+1; i++) {
+    //     for (int icrm=0; icrm<ncrms; icrm++) {
+    parallel_for( SimpleBounds<3>(nzm,nwx,ncrms) , YAKL_LAMBDA (int k, int i, int icrm) {
+      int ii = i + 2*(filter_wn_max+1) ;
+      fft_out(k,0,ii,icrm) = 0.0;
+    });
+  }
+
+  //----------------------------------------------------------------------------
+  // Backward Fourier transform
+
+  if (RUN3D) {
+    // for (int k=0; k<nzm; k++) {
+    //   for (int i=0; i<nx+1; i++) {
+    //     for (int icrm=0; icrm<ncrms; icrm++) {
+    parallel_for( SimpleBounds<3>(nzm,nx+1,ncrms) , YAKL_LAMBDA (int k, int i, int icrm) {
+      real ftmp[ny+2]; real tmp [ny];
+      for(int j=0; j<ny+2; j++) { ftmp[j] = fft_out(k,j,i,icrm); }
+      ffty.inverse(ftmp,tmp);
+      for(int j=0; j<ny  ; j++) { fft_out(k,j,i,icrm) = ftmp[j]; } 
+    });
+  }
+
+  // for (int k=0; k<nzm; k++) {
+  //   for (int j=0; i<ny; i++) {
+  //     for (int icrm=0; icrm<ncrms; icrm++) {
+  parallel_for( SimpleBounds<3>(nzm,ny,ncrms) , YAKL_LAMBDA (int k, int j, int icrm) {
+    real ftmp[nx+2]; real tmp [nx];
+    for(int i=0; i<nx+2; i++) { ftmp[i] = fft_out(k,j,i,icrm); }
+    fftx.inverse(ftmp,tmp);
+    for(int i=0; i<nx  ; i++) { f_out(k,j,i,icrm) = ftmp[i]; }
+  });
+
+}
+
+//==============================================================================
+//==============================================================================
+
+void VT_diagnose() {
+  auto &t            = :: t;
+  auto &qv           = :: qv;
+  auto &qcl          = :: qcl;
+  auto &qci          = :: qci;
+  auto &factor_xy    = :: factor_xy;
+  auto &t_vt_pert    = :: t_vt_pert;
+  auto &q_vt_pert    = :: q_vt_pert;
+  auto &t_vt         = :: t_vt;
+  auto &q_vt         = :: q_vt;
+  auto &ncrms        = :: ncrms;
+
+  // local variables
+  real2d t_mean("t_mean", nzm, ncrms);
+  real2d q_mean("q_mean", nzm, ncrms);
+  real4d tmp_t("tmp_t", nzm, ny, nx, ncrms);
+  real4d tmp_q("tmp_q", nzm, ny, nx, ncrms);
+
+#if defined(MMF_VT_KMAX)
+  int filter_wn_max = MMF_VT_KMAX;
+#else
+  int filter_wn_max = 1;
+#endif
+
+  //----------------------------------------------------------------------------
+  // calculate horizontal mean
+  //----------------------------------------------------------------------------
+  // do k = 1,nzm
+  //   do icrm = 1,ncrms
+  parallel_for( SimpleBounds<2>(nzm,ncrms) , YAKL_LAMBDA (int k, int icrm) {
+      t_mean(k,icrm) = 0.0;
+      q_mean(k,icrm) = 0.0;
+      t_vt(k,icrm) = 0.0;
+      q_vt(k,icrm) = 0.0;
+  });
+
+  // do k = 1,nzm
+  //  do j = 1,ny
+  //    do i = 1,nx
+  //      do icrm = 1,ncrms
+  parallel_for( SimpleBounds<4>(nzm,ny,nx,ncrms) , YAKL_LAMBDA (int k, int j, int i, int icrm) {
+    real t_tmp = t(k,j+offy_s,i+offx_s,icrm);
```

Comment:
Rename these because of confusion with tmp_t and tmp_q

Reply:
If you want, you can inline these temps. `yakl::atomicAdd` doesn't need the 2nd argument to be stored in a temp like Fortran's `!$acc atomic` does

Reply:
done.